compiler/src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -221,28 +221,26 @@ object Settings {
      *
      *  to get their arguments.
      */
-    def processArguments(state: ArgsSummary, processAll: Boolean, skipped: List[String]): ArgsSummary = {
+    final def processArguments(state: ArgsSummary, processAll: Boolean, skipped: List[String]): ArgsSummary =
```

Comment:
There are also non-tailrec calls for dash-options, as opposed to args. It would be nice if I could call
```
loop(): @tailrec
```
but I'm so lazy I didn't try that to see if it works. I assume it doesn't because I haven't seen a billboard advertising it.

Reply:
I see, in that case please add a comment mentioning that

Reply:
Wound up fixing the other loop and did some scouting.